iceoryx_posh/source/roudi/process_manager.cpp

```diff
@@ -45,29 +45,37 @@ ProcessManager::ProcessManager(RouDiMemoryInterface& roudiMemoryInterface,
     , m_portManager(portManager)
     , m_compatibilityCheckLevel(compatibilityCheckLevel)
 {
+    bool fatalError{false};
+
     auto maybeSegmentManager = m_roudiMemoryInterface.segmentManager();
     if (!maybeSegmentManager.has_value())
     {
         LogFatal() << "Invalid state! Could not obtain SegmentManager!";
-        std::terminate();
+        fatalError = true;
     }
     m_segmentManager = maybeSegmentManager.value();
 
     auto maybeIntrospectionMemoryManager = m_roudiMemoryInterface.introspectionMemoryManager();
     if (!maybeIntrospectionMemoryManager.has_value())
     {
         LogFatal() << "Invalid state! Could not obtain MemoryManager for instrospection!";
-        std::terminate();
+        fatalError = true;
     }
     m_introspectionMemoryManager = maybeIntrospectionMemoryManager.value();
 
     auto maybeMgmtSegmentId = m_roudiMemoryInterface.mgmtMemoryProvider()->segmentId();
     if (!maybeMgmtSegmentId.has_value())
     {
         LogFatal() << "Invalid state! Could not obtain SegmentId for iceoryx management segment!";
-        std::terminate();
+        fatalError = true;
     }
     m_mgmtSegmentId = maybeMgmtSegmentId.value();
+
+    if (fatalError)
```

Comment:
I see your point, but IMHO the maintenance effort is not much higher, hence I'll stick with the current approach as this will be refactored relatively soon anyway.

Reply:
it's your decision, but keep in mind, nothing is more definitive than the temporary ;)